tests/test_factory_constructors.cpp

```diff
@@ -20,12 +20,12 @@ class TestFactory1 {
     TestFactory1() : value("(empty)") { print_default_created(this); }
     TestFactory1(int v) : value(std::to_string(v)) { print_created(this, value); }
     TestFactory1(std::string v) : value(std::move(v)) { print_created(this, value); }
+public:
+    std::string value;
     TestFactory1(TestFactory1 &&) = delete;
```

Comment:
It complained that deleted functions need to be public per the spec so I fixed it.

Reply:
Specifically:
```
pybind11/tests/test_factory_constructors.cpp:23:5: error: deleted member function should be public [modernize-use-equals-delete,-warnings-as-errors]
    TestFactory1(TestFactory1 &&) = delete;
    ^
pybind11/tests/test_factory_constructors.cpp:24:5: error: deleted member function should be public [modernize-use-equals-delete,-warnings-as-errors]
    TestFactory1(const TestFactory1 &) = delete;
    ^
pybind11/tests/test_factory_constructors.cpp:25:19: error: deleted member function should be public [modernize-use-equals-delete,-warnings-as-errors]
    TestFactory1 &operator=(TestFactory1 &&) = delete;
pybind11/tests/test_factory_constructors.cpp:26:19: error: deleted member function should be public [modernize-use-equals-delete,-warnings-as-errors]
    TestFactory1 &operator=(const TestFactory1 &) = delete;
```